pkg/agent/runner.go

```diff
@@ -257,12 +257,30 @@ func (r *Runner) Run(ctx context.Context, logger *zap.Logger, vmInfoUpdated util
 			l,
 			"system",
 			func() *core.SystemMetrics { return new(core.SystemMetrics) },
+			func() bool { return true },
 			r.global.config.Metrics.System,
 			func(metrics *core.SystemMetrics, withLock func()) {
 				ecwc.Updater().UpdateSystemMetrics(*metrics, withLock)
 			},
 		)
 	})
+	r.spawnBackgroundWorker(ctx, logger, "get LFC metrics", func(c context.Context, l *zap.Logger) {
+		getMetricsLoop(
+			r,
+			c,
+			l,
```

Comment:
ctx2, logger2?

Otherwise, `r, c, l` is confusing IMO

Reply:
That's fair. All of the other functions in that area use the same naming — I'm hesitant to change these without changing those as well. The reason they're like that currently is because I thought shadowing would be confusing, and it'd be too verbose to write out `ctx2` and `logger2` in all of the usages.

What do you think?

Reply:
Added this as a new commit (to be kept separate after rebasing) in efdda7c819bbfe1a5178b884d9803438b5cebba1.